src/kernel/MomentumMassElemKernel.C

```diff
@@ -61,6 +60,8 @@ MomentumMassElemKernel<AlgTraits>::MomentumMassElemKernel(
 
   MasterElement* meSCV = sierra::nalu::MasterElementRepo::get_volume_master_element(AlgTraits::topo_);
 
+  // Initialize ipNodeMap_
+  get_scv_ipnodemap<AlgTraits>(meSCV->ipNodeMap(), ipNodeMap_);
```

Comment:
@alanw0 here is my proposed design for transitioning `ipNodeMap_` from a pointer to Kokkos::View, and follows the logic that was already there for shape functions.

Reply:
I thought the idea was that the ipNodeMap would be a static POD array or a Kokkos view in the master element so that it would be available on the GPU without copying it to a View and then deepcopying the view.

Reply:
I don't think there's any issue with that.  AlgTraits to MasterElement types aren't exactly one-to-one, but we need to write separate kernels for different discretizations anyway so we'd know if we're working with a Hex8FEM or Hex8CVFEM master element.

Reply:
@overfelt I see what you are saying, `execute()` is entered for each element, so I didn't want to ask for ME instance everytime. However, it makes no difference since you have them already stored in a repo and you're returning the same instance even on GPU, so I think we can remove `ipNodeMap_` altogether. I assume this is where you were going?

Reply:
Yes,
meSCV=sierra::nalu::MasterElementRepo::get_volume_master_element<AlgTraits>();
will return a master element pointer for use on the GPU and ipNodeMap() can be called to return the array pointer that will also be on the GPU.  Since the master element is stored as a pointer on the class it does not have to be allocated every time execute() is called.

Reply:
@overfelt OK. Got it. I assume we will want the same thing for 

```
  /// Shape functions
  AlignedViewType<DoubleType[AlgTraits::numScvIp_][AlgTraits::nodesPerElement_]> v_shape_function_ {"view_shape_func"};
```

Reply:
Yes, pass v_shape_function_ to:
template<typename ViewType> KOKKOS_FUNCTION void HexSCV::shape_fcn(ViewType &shpfc)
and it will fill it in just as the similar call does on the CPU.

Reply:
Note that for some reason the master elements do not keep these shape function evaluations around even though they are constant.  Maybe we should change that?

Reply:
I was just going to say! Not sure why we compute it every time

Reply:
For the low order elements, the cost of computing the shape functions is in the noise, so we didn't switch to saving them off.  But with the GPU, it makes sense just to save them.   The bigger one is the shape function derivatives which can also be saved (though also not performance critical).

Reply:
It appears that in the kernel design, the shape functions were saved off, just not in the MasterElement instance but in the kernel instance.